src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,212 @@
+import classNames from 'classnames';
+import { useContext, useState } from 'react';
+import * as commentAPI from '../../api/commentAPI';
+import { PostContext } from '../../context/PostContext';
+import { Comment } from '../../types/Comment';
+
+export const NewCommentForm: React.FC = () => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [text, setText] = useState('');
+  const [textError, setTextError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const {
+    selectedPost,
+    setComments,
+    setErrorMessage,
+  } = useContext(PostContext);
+
+  const allReset = () => {
+    setName('');
+    setNameError(false);
+    setEmail('');
+    setEmailError(false);
+    setText('');
+    setTextError(false);
+  };
+
+  const successReset = () => {
+    setText('');
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!name) {
+      setNameError(true);
+    }
+
+    if (!email) {
+      setEmailError(true);
+    }
+
+    if (!text) {
+      setTextError(true);
+    }
+
+    if (!name || !email || !text) {
+      return;
+    }
+
+    setLoading(true);
+
+    const newComment = {
+      postId: selectedPost?.id,
+      name,
+      email,
+      body: text,
+    };
+
+    commentAPI.createComment(newComment as Comment)
+      .then(comment => setComments(
+        currentComments => [...currentComments, comment],
+      ))
+      .catch(() => setErrorMessage('Something went wrong!'))
+      .finally(() => setLoading(false));
+
+    successReset();
+  };
+
+  return (
+    <form data-cy="NewCommentForm" onSubmit={handleFormSubmit}>
+      <div className="field" data-cy="NameField">
+        <label className="label" htmlFor="comment-author-name">
+          Author Name
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
```

Comment:
check your inputs and text area and you can see, you can combine it in 1 component like `TextField` and reuse it. You need to pass right props to show text area or input